test/Bridges/Constraint/semi_to_binary.jl

```diff
@@ -0,0 +1,135 @@
+        bridged_mock,
+        MOI.ListOfConstraintIndices{MOI.SingleVariable,
+                                    MOI.Semicontinuous{Float64}}()))
+
+    test_delete_bridge(bridged_mock, ci, 2, (
+        (MOI.SingleVariable, MOI.EqualTo{Float64}, 1),
+        (MOI.SingleVariable, MOI.ZeroOne, 0),
+        (MOI.SingleVariable, MOI.Integer, 0),
+        (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}, 0),
+        (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}, 1),
+        ))
+
+    MOIU.set_mock_optimize!(mock,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 0.0)
+            MOIU.mock_optimize!(mock, [0.0, 0.0, 0.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.0)
+            MOIU.mock_optimize!(mock, [2.0, 1.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.0)
+            MOIU.mock_optimize!(mock, [2.0, 2.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 3.0)
+            MOIU.mock_optimize!(mock, [3.0, 2.5, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 3.0)
+            MOIU.mock_optimize!(mock, [3.0, 3.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> MOI.set(mock, MOI.TerminationStatus(), MOI.INFEASIBLE)
+    )
+    MOIT.semiinttest(bridged_mock,config)
+
+    ci = first(MOI.get(
+        bridged_mock,
+        MOI.ListOfConstraintIndices{MOI.SingleVariable,
+                                    MOI.Semiinteger{Float64}}()))
+
+    @test MOI.get(bridged_mock, MOI.ConstraintPrimal(), ci) == 3
+    new_set = MOI.Semiinteger{Float64}(19.0, 20.0)
+    MOI.set(bridged_mock, MOI.ConstraintSet(), ci, new_set)
+    @test MOI.get(bridged_mock, MOI.ConstraintSet(), ci) == new_set
+
+    @testset "$attr" for attr in [MOI.ConstraintPrimalStart(),]
+        @test MOI.supports(bridged_mock, attr, typeof(ci))
+        value = 2.0
+        MOI.set(bridged_mock, attr, ci, value)
+        @test MOI.get(bridged_mock, attr, ci) ≈ value
+    end
+
+    test_delete_bridge(bridged_mock, ci, 2, (
+        (MOI.SingleVariable, MOI.EqualTo{Float64}, 1),
+        (MOI.SingleVariable, MOI.ZeroOne, 0),
+        (MOI.SingleVariable, MOI.Integer, 0),
+        (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}, 0),
+        (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}, 1),
+        ))
+
+
+    s = """
+    variables: x, y
+    cy: y == 4.0
+    cx: x in Semiinteger(2.0, 3.0)
+    cxy: x + -1.0y >= 0.0
+    minobjective: x
+    """
+    model = MOIU.Model{Float64}()
+    MOIU.loadfromstring!(model, s)
+    MOI.empty!(bridged_mock)
+    @test MOI.is_empty(bridged_mock)
+    MOIU.loadfromstring!(bridged_mock, s)
+    MOIU.test_models_equal(bridged_mock, model, ["x", "y"], ["cy", "cx", "cxy"])
```

Comment:
You should also do `test_model_equals(mock, ...)` to test that the model was bridged into the expected model.

Reply:
What do I do with Names in the `mock`?
They don't exist, right?

Reply:
You need to set them, see
https://github.com/JuliaOpt/MathOptInterface.jl/blob/5b25a050d99efb1a57a796d4ca2ffc5cdcf44e17/test/Bridges/Constraint/geomean.jl#L30-L37